chirp/drivers/kenwood_d7.py

```diff
@@ -13,332 +13,3404 @@
+            index = self._get_index(element)
+            pvdl = "PV %dL" % index
+            pvdu = "PV %dU" % index
+            if element[pvdl].changed() or element[pvdu].changed():
+                lower = element[pvdl].value.get_value()
+                upper = element[pvdu].value.get_value()
+                args = "%05d,%05d" % (lower, upper)
+                self._radio._kenwood_set(element.get_name(), args)
+                element[pvdl].value._has_changed = False
+                element[pvdu].value._has_changed = False
+
+
+class KenwoodD7Family(chirp_common.LiveRadio):
+    VENDOR = "Kenwood"
+    MODEL = ""
+    NEEDS_COMPAT_SERIAL = False
     HARDWARE_FLOW = False
 
-    _kenwood_split = True
-    _upper = 199
-
-    _BEP_OPTIONS = ["Off", "Key", "Key+Data", "All"]
-    _POSC_OPTIONS = ["Off Duty", "Enroute", "In Service", "Returning",
-                     "Committed", "Special", "Priority", "Emergency"]
-
-    _SETTINGS_OPTIONS = {
-        "BAL": ["4:0", "3:1", "2:2", "1:3", "0:4"],
-        "BEP": None,
-        "BEPT": ["Off", "Mine", "All New"],  # D700 has fourth "All"
-        "DS": ["Data Band", "Both Bands"],
-        "DTB": ["A", "B"],
-        "DTBA": ["A", "B", "A:TX/B:RX"],  # D700 has fourth A:RX/B:TX
-        "DTX": ["Manual", "PTT", "Auto"],
-        "ICO": ["Kenwood", "Runner", "House", "Tent", "Boat", "SSTV",
-                "Plane", "Speedboat", "Car", "Bicycle"],
-        "MNF": ["Name", "Frequency"],
-        "PKSA": ["1200", "9600"],
-        "POSC": None,
-        "PT": ["100ms", "200ms", "500ms", "750ms",
-               "1000ms", "1500ms", "2000ms"],
-        "SCR": ["Time", "Carrier", "Seek"],
-        "SV": ["Off", "0.2s", "0.4s", "0.6s", "0.8s", "1.0s",
-               "2s", "3s", "4s", "5s"],
-        "TEMP": ["F", "C"],
-        "TXI": ["30sec", "1min", "2min", "3min", "4min", "5min",
-                "10min", "20min", "30min"],
-        "UNIT": ["English", "Metric"],
-        "WAY": ["Off", "6 digit NMEA", "7 digit NMEA", "8 digit NMEA",
-                "9 digit NMEA", "6 digit Magellan", "DGPS"],
-    }
-
-    def get_features(self):
-        rf = chirp_common.RadioFeatures()
-        rf.has_settings = True
-        rf.has_dtcs = False
-        rf.has_dtcs_polarity = False
-        rf.has_bank = False
-        rf.has_mode = True
-        rf.has_tuning_step = False
-        rf.can_odd_split = True
-        rf.valid_duplexes = ["", "-", "+", "split"]
-        rf.valid_modes = list(MODES.values())
-        rf.valid_tmodes = ["", "Tone", "TSQL"]
-        rf.valid_characters = \
-            chirp_common.CHARSET_ALPHANUMERIC + "/.-+*)('&%$#! ~}|{"
-        rf.valid_name_length = 7
-        rf.valid_tuning_steps = STEPS
-        rf.memory_bounds = (0, self._upper)
-        return rf
+    _ARG_DELIMITER = " "
+    _BAUDS = [9600]
+    _CMD_DELIMITER = "\r"
+    _DISABLE_AI_CMD = ('AI', '0')
+    _DUPLEX = {0: "", 1: "+", 2: "-"}
+    _HAS_NAME = True
+    _LOCK = threading.Lock()
```

Comment:
Is this necessary? The RadioThread should be handling dispatching one request at a time to the driver...

Reply:
I only have it there because it was in there before (and is still in kenwood_live).  Happy to remove it.

Reply:
Ah, sorry, it's been a long time since I looked at this stuff. Should be good to remove it.

Reply:
Turns out it's needed so I can click the settings tab while the memories are loading.